yorkie/src/main/kotlin/dev/yorkie/document/crdt/RgaTreeList.kt

```diff
@@ -0,0 +1,242 @@
+package dev.yorkie.document.crdt
+
+import androidx.annotation.VisibleForTesting
+import dev.yorkie.document.time.TimeTicket
+import dev.yorkie.document.time.TimeTicket.Companion.InitialTimeTicket
+import dev.yorkie.util.SplayTreeSet
+
+/**
+ * RgaTreeList is replicated growable array.
+ */
+internal class RgaTreeList {
+    val dummyHead = RgaTreeListNode(Primitive.of(1, InitialTimeTicket)).apply {
+        value.removedAt = InitialTimeTicket
+    }
+    var last: RgaTreeListNode = dummyHead
+        private set
+
+    private val nodeMapByIndex = SplayTreeSet<CrdtElement> { if (it.isRemoved) 0 else 1 }.apply {
+        insert(dummyHead.value)
+    }
+    private val nodeMapByCreatedAt = mutableMapOf<String, RgaTreeListNode>().apply {
```

Comment:
We can use TimeTicket directly instead of `String` here.

```suggestion
    private val nodeMapByCreatedAt = mutableMapOf<TimeTicket, RgaTreeListNode>().apply {
```